mne/time_frequency/tests/test_csd.py

```diff
@@ -367,28 +367,30 @@ def _test_csd_matrix(csd):
 
 def _test_fourier_multitaper_parameters(epochs, csd_epochs, csd_array):
     """Parameter tests for csd_*_fourier and csd_*_multitaper."""
-    raises(ValueError, csd_epochs, epochs, fmin=20, fmax=10)
-    raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
-           epochs.tmin, fmin=20, fmax=10)
-    raises(ValueError, csd_epochs, epochs, fmin=20, fmax=20.1)
-    raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
-           epochs.tmin, fmin=20, fmax=20.1)
-    raises(ValueError, csd_epochs, epochs, tmin=0.15, tmax=0.1)
-    raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
-           epochs.tmin, tmin=0.15, tmax=0.1)
-    raises(ValueError, csd_epochs, epochs, tmin=-1, tmax=10)
-    raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
-           epochs.tmin, tmin=-1, tmax=10)
-    raises(ValueError, csd_epochs, epochs, tmin=10, tmax=11)
-    raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
-           epochs.tmin, tmin=10, tmax=11)
+    with warnings.catch_warnings(record=True):  # deprecation
+        raises(ValueError, csd_epochs, epochs, fmin=20, fmax=10)
+        raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
+               epochs.tmin, fmin=20, fmax=10)
+        raises(ValueError, csd_epochs, epochs, fmin=20, fmax=20.1)
+        raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
+               epochs.tmin, fmin=20, fmax=20.1)
+        raises(ValueError, csd_epochs, epochs, tmin=0.15, tmax=0.1)
+        raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
+               epochs.tmin, tmin=0.15, tmax=0.1)
+        raises(ValueError, csd_epochs, epochs, tmin=-1, tmax=10)
+        raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
+               epochs.tmin, tmin=-1, tmax=10)
+        raises(ValueError, csd_epochs, epochs, tmin=10, tmax=11)
+        raises(ValueError, csd_array, epochs._data, epochs.info['sfreq'],
+               epochs.tmin, tmin=10, tmax=11)
```

Comment:
perhaps parts of this can be done with looping? I think that can make the structure clearer.